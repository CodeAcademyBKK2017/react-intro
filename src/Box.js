import React, { Component } from 'react';
import Row from './Row';

class Box extends Component {
  render() {
    return (
        <div>
            <Row/> <Row/> <Row/>
        </div>
    );
  }
}

export default Box;
