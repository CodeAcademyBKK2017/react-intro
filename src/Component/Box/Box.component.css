.Box{
    width: 300px;
    border: red solid;
    padding: 15px;
}
