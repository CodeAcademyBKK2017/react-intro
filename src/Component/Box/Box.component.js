import React ,{Component}from 'react';
import Row from '../Row/Row.component'
// class Hello extends Component{
//     cellclickHandler=(celData)=>{
//         return()=>{
//             console.log('cellclickHandler',celData);
//         }
        
//     }
//     render(){
    
//         return(
//             <div>
//                 <Row class='row' data ={this.props.data[0]} cellclickHandler={this.cellclickHandler}/> 
//                 <Row class='row' data ={this.props.data[1]} cellclickHandler={this.cellclickHandler}/> 
//                 <Row class='row' data ={this.props.data[2]} cellclickHandler={this.cellclickHandler}/> 
//             </div> 
//         )
//     }
// }
const Hello  =(props)=>{
    const cellclickHandler=(celData) => console.log('cellclickHandler',celData);
            
    return(
        <div>
            <Row class='row' data ={props.data[0]} cellclickHandler={cellclickHandler}/> 
            <Row class='row' data ={props.data[1]} cellclickHandler={cellclickHandler}/> 
            <Row class='row' data ={props.data[2]} cellclickHandler={cellclickHandler}/> 
        </div> 
    )
}
export default Hello;
