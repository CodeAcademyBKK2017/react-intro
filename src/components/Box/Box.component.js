import PropTypes from 'prop-types';
import React, {Component} from 'react';
import Row from '../Row/Row.component';

class Box extends Component {

  cellclickHandler (cellValue) {

    return () => {};

  }

  createRow = (rowData) => <Row cellData={rowData.items}
    key={rowData.id} cellclickHandler={this.cellclickHandler} />

  render () {

    const {boxData} = this.props;
    const rows = boxData.map(this.createRow);
    return (
      <div>
        {rows}
      </div>
    );

  }

}
Box.propTypes = {'boxData': PropTypes.array.isRequired};
Box.defaultProps = {'boxData': []};
export default Box;
