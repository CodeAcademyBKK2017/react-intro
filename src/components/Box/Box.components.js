import React, { Component } from 'react';
import Row from '../Row/Row.components';

class Box extends Component {

    // cellClickHandler = () => { // class preproties for es7 feature
    //     console.log('cell clicked');
    // }
    cellClickHandler(){ //class method
        console.log('cell clicked');
    }
    render() {
        // this.props.rowData
        return (
            <div>
                <Row cellData={  this.props.rowData[0] } cellClickHandler={this.cellClickHandler}/> 
                <Row cellData={  this.props.rowData[1] } cellClickHandler={this.cellClickHandler}/> 
                <Row cellData={  this.props.rowData[2] } cellClickHandler={this.cellClickHandler}/>
            </div>
        );
  }
}

export default Box;
