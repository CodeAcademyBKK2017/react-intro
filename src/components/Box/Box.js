import React, { Component } from 'react';
import Row from '../Row/Row';

class Box extends Component {
  render() {
    return (
        <div>
            <Row/> <Row/> <Row/>
        </div>
    );
  }
}

export default Box;
