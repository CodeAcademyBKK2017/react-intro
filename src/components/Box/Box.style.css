.box {
    display: flex;
    border: 2px solid red;
    padding: 10vw;
    align-content: center;
    align-items: center;
}