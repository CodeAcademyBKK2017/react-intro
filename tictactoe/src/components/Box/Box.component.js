import React from 'react';

import Row from '../Row/Row.component'

class Box extends React.Component {
    cellClickHandler = (cellData) => () => {
        console.log('cell clicked', cellData);
    }
    render() {
        const rowLists = this.props.boxData.map((rowData, index) => <Row key={index} rowData={rowData} cellClickHandler={this.cellClickHandler} />);
        return (
            <div>
                {rowLists}
            </div>
        );
    }
}

export default Box;
