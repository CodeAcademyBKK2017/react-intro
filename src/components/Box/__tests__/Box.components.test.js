import renderer from 'react-test-renderer';
import React from 'react';
import Box from '../Box.components';

test('Box: snapshot test', () => {
    const titacto = 
    [
      ['O', 'X', 'O'],
      ['X', 'O', 'O'],
      ['X', 'X', 'O']
    ];
    const snapshot = renderer.create(<Box rowData={ titacto }/>);
    expect(snapshot).toMatchSnapshot();
});