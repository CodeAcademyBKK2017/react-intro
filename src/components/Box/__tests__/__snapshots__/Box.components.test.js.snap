// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box: snapshot test 1`] = `
<div>
  <div
    style={
      Object {
        "marginTop": "44px",
      }
    }
  >
    <div
      className="cell-box"
      onClick={[Function]}
    >
      O
    </div>
     
    <div
      className="cell-box"
      onClick={[Function]}
    >
      X
    </div>
     
    <div
      className="cell-box"
      onClick={[Function]}
    >
      O
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "44px",
      }
    }
  >
    <div
      className="cell-box"
      onClick={[Function]}
    >
      X
    </div>
     
    <div
      className="cell-box"
      onClick={[Function]}
    >
      O
    </div>
     
    <div
      className="cell-box"
      onClick={[Function]}
    >
      O
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "44px",
      }
    }
  >
    <div
      className="cell-box"
      onClick={[Function]}
    >
      X
    </div>
     
    <div
      className="cell-box"
      onClick={[Function]}
    >
      X
    </div>
     
    <div
      className="cell-box"
      onClick={[Function]}
    >
      O
    </div>
  </div>
</div>
`;
